src/extensions/scratch3_wedo2/index.js

```diff
@@ -549,20 +539,11 @@ class WeDo2 {
      * @private
      */
     _send (uuid, message) {
-        if (!this.getPeripheralIsConnected()) return;
-        if (this._sending) return;
+        if (!this.getPeripheralIsConnected()) return Promise.resolve();
```

Comment:
oooh, what is Promise.resolve() doing here (and in line 544?)

Reply:
Good question! The JSDOC indicates that the function should return a promise, but we were just returning null from it when we needed to bail out, previously. The rate limiter change revealed a bug where another function (_stopAll) used a then() function on another function that returned the promise from _send(). In the case where the message is dropped because of rate limiting, the _send function was not returning a promise, and the attempt to use then() threw an error. 

I suppose this means we should reconsider whether to use promise chaining at all here!